data_processors/pipeline/lambdas/bcl_convert.py

```diff
@@ -193,6 +198,66 @@ def get_settings_by_instrument_type_assay(instrument, sample_type, assay):
     return {}
 
 
+# TODO: extract into utils?
+def get_library_id_from_sample_name(sample_name: str):
+    # format: samplename_libraryid_extension
+    # we are only interested in the library ID
+    fragments = sample_name.split("_")
+    # if there is an extension, the library ID is the second to last fragment
+    if "_topup" in sample_name or "_rerun" in sample_name:
+        return fragments[-2]
+    # if not, then the library ID is the last fragment
+    return fragments[-1]
+
+
+# TODO: extract SampleSheet handling/utils into separate module?
+def get_sample_names_from_samplesheet(gds_volume: str, samplesheet_path: str) -> List[str]:
+    if not samplesheet_path.startswith(os.path.sep):
+        samplesheet_path = os.path.sep + samplesheet_path
+    logger.info(f"Extracting sample names from gds://{gds_volume}{samplesheet_path}")
+
+    ntf: NamedTemporaryFile = gds.download_gds_file(gds_volume, samplesheet_path)
+    if ntf is None:
+        reason = f"Abort extracting metadata process. " \
+                 f"Can not download sample sheet from GDS: gds://{gds_volume}{samplesheet_path}"
+        logger.error(reason)
+        raise ValueError(reason)
+
+    logger.info(f"Local sample sheet path: {ntf.name}")
+    sample_names = set()
+    with closing(ntf) as f:
+        samplesheet = SampleSheet(f.name)
+        for sample in samplesheet:
+            sample_names.add(get_library_id_from_sample_name(sample.Sample_ID))
```

Comment:
Should not deduce to `library_id` just yet. It should just collect Sample_ID as in SampleSheet i.e.

```
sample_names.add(sample.Sample_ID)
```

Reply:
Agreed. We can filter the library_id out later. I reverted that logic, but seem to have forgotten to remove the call here.